src/store/sagas/positions.ts

```diff
@@ -254,23 +97,27 @@ function* handleInitPositionWithAZERO(action: PayloadAction<InitPositionData>):
       true
     )
 
-    let azeroAmount = 0n
-    if (tokenX === TESTNET_WAZERO_ADDRESS) {
-      azeroAmount = azeroBalance > xAmountWithSlippage ? xAmountWithSlippage : azeroBalance
-    } else {
-      azeroAmount = azeroBalance > yAmountWithSlippage ? yAmountWithSlippage : azeroBalance
+    if (
+      (tokenX === wazeroAddress && tokenXAmount !== 0n) ||
+      (tokenY === wazeroAddress && tokenYAmount !== 0n)
+    ) {
+      let azeroAmount = 0n
+      if (tokenX === wazeroAddress) {
+        azeroAmount = azeroBalance > xAmountWithSlippage ? xAmountWithSlippage : azeroBalance
+      } else {
+        azeroAmount = azeroBalance > yAmountWithSlippage ? yAmountWithSlippage : azeroBalance
+      }
+
+      const depositTx = wazero.depositTx(azeroAmount, WAZERO_DEPOSIT_OPTIONS)
+      txs.push(depositTx)
     }
-    const depositTx = wazero.depositTx(azeroAmount, WAZERO_DEPOSIT_OPTIONS)
-    txs.push(depositTx)
 
     const XTokenTx = psp22.approveTx(invAddress, xAmountWithSlippage, tokenX, PSP22_APPROVE_OPTIONS)
```

Comment:
Why you approving both token x and token y?

Reply:
you have to approve both tokens when creating position